DxDispatch/src/dxdispatch/Logging.cpp

```diff
@@ -1,26 +1,82 @@
 #include "pch.h"
 #include <iostream>
 
-void LogInfo(std::string_view msg)
+HRESULT DxDispatchConsoleLogger::QueryInterface(
+    REFIID riid,
+    _COM_Outptr_ void ** ppvObject)
+{
+    *ppvObject = nullptr;
+    if ((IID_IUnknown == riid) ||
+        __uuidof(IDxDispatchLogger) == riid)
+    {
+        *ppvObject = static_cast<IDxDispatchLogger*>(this);
+    }
+    else
+    {
+        return E_NOTIMPL;
+    }
+    AddRef();
+    return S_OK;
+}
+
+ULONG DxDispatchConsoleLogger::AddRef(void)
+{
+    return ++m_refCount;
```

Comment:
This would need InterlockedIncrement/Decrement to make addref/release thread safe. It's probably better to inherit from WRL::Base to take care of implementing IUnknown. I believe this helper is available in the WSL headers:

https://github.com/microsoft/DirectX-Headers/blob/c0f2af8af649833910786f38eb79cc333c77b1a6/include/wsl/wrladapter.h#L760

Reply:
I think it should be fine since m_refCount is atomic variable and WSL is using same variable type, interlocked commands are not supported in linux